core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -188,39 +189,33 @@ private[spark] class CoarseMesosSchedulerBackend(
           taskIdToSlaveId(taskId) = slaveId
           slaveIdsWithExecutors += slaveId
           coresByTaskId(taskId) = cpusToUse
-          val task = MesosTaskInfo.newBuilder()
+          // Gather cpu resources from the available resources and use them in the task.
+          val (remainingResources, taskResources) =
+            partitionResources(offer.getResourcesList, "cpus", cpusToUse)
```

Comment:
(@tnachen follow up question: why do we allocate CPU resources then memory resources with whatever is remaining?)